Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -214,27 +362,39 @@ static RefPtr<SharedBuffer> parseResponseMessage(const String& message)
     return SharedBuffer::create(message.characters8() + offset, message.sizeInBytes() - offset);
 }
 
-Ref<CDMInstanceOpenCDM::Session> CDMInstanceOpenCDM::Session::create(const media::OpenCdm& source, Ref<WebCore::SharedBuffer>&& initData)
+Ref<CDMInstanceOpenCDM::Session> CDMInstanceOpenCDM::Session::create(CDMInstanceOpenCDM::AccessorHandle source, const String& keySystem, const char* type, Ref<WebCore::SharedBuffer>&& initData, LicenseType licenseType, Ref<WebCore::SharedBuffer>&& customData)
 {
-    return adoptRef(*new Session(source, WTFMove(initData)));
+    return adoptRef(*new Session(source, keySystem, type, WTFMove(initData), licenseType, WTFMove(customData)));
 }
 
-CDMInstanceOpenCDM::Session::Session(const media::OpenCdm& source, Ref<WebCore::SharedBuffer>&& initData)
-    : m_session(source)
-    , m_initData(WTFMove(initData))
+CDMInstanceOpenCDM::Session::Session(CDMInstanceOpenCDM::AccessorHandle source, const String& keySystem, const char* mimeType, Ref<WebCore::SharedBuffer>&& initData, LicenseType licenseType, Ref<WebCore::SharedBuffer>&& customData)
+    : m_initData(WTFMove(initData))
 {
-    // FIXME: This can be a source of overflows and other big errors. This should be fixed in the framework and then fixed here.
-    uint8_t temporaryURL[1024];
-    uint16_t temporaryURLLength = sizeof(temporaryURL);
+    if (source == InvalidAccessorHandle)
+        return;
 
-    std::string message;
-    m_session.GetKeyMessage(message, temporaryURL, temporaryURLLength);
+    auto& sessonCallbacks = OCDMSessionCallbacksNotifier::singleton();
 
-    if (message.empty() || !temporaryURLLength)
+    OpenCDMSession* session = nullptr;
+    opencdm_create_session(source, keySystem.utf8().data(), openCDMLicenseType(licenseType), mimeType, reinterpret_cast<const uint8_t*>(m_initData->data()), m_initData->size(),
```

Comment:
From an API pov, it might be nice to have some sort of struct encapsulating these parameters, anything taking more 4 in my book is a candidate for this, very hard to see what's going on.

Reply:
I agree. A struct with these grouped would be better but I approached it like OCDM is a third party library I have no control over in order to avoid changing the stuff all over the place. The API is as it is (and it wasn't that much better with C++ one too).
Would it be an option to leave it as is for time being, try to make it better in OCDM and change that separately one that happens?

Reply:
Sure

Reply:
Thanks